connectors/sources/sharepoint.py

```diff
@@ -284,7 +284,25 @@ async def _api_call(self, url_name, url="", **url_kwargs):
                         yield await result.json()
                     break
             except Exception as exception:
```

Comment:
Should you change this `except Exception` to handle the specific exception that means that request was throttled?

Reply:
we are handling multiple exceptions inside this exception and if handle specific exception here then logic for retry will be duplicated in all exceptions.

Reply:
You can provide private methods for exception handling shared logic, the way it's done now is really hard to read and understand